scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -128,12 +140,76 @@ public SchedulingService(SchedulingInfrastructure infrastructure, SchedulerState
 
         pinger = new NodePingThread(this);
         pinger.start();
+
+        if (PASchedulerProperties.SCHEDULER_AUTOMATIC_REMOVED_JOB_DELAY.getValueAsInt() > 0) {
+            startHouseKeeping(this.jobsToDeleteFromDB, this.infrastructure);
+        }
+    }
+
+    public void startHouseKeeping(final ConcurrentLinkedQueue<JobId> jobsQueue,
+            final SchedulingInfrastructure infrastructure) {
+        houseKeepingScheduler = new Scheduler();
+        String cronExpr = "* * * * *";
+        if (PASchedulerProperties.SCHEDULER_AUTOMATIC_REMOVED_JOB_CRON_EXPR.isSet()) {
+            cronExpr = PASchedulerProperties.SCHEDULER_AUTOMATIC_REMOVED_JOB_CRON_EXPR.getValueAsString();
+        }
+        houseKeepingScheduler.schedule(cronExpr, new Runnable() {
```

Comment:
I think it would be clearer to split this long method into several smaller ones, maybe create an inner class